### PR TITLE
Pre-built dbt Tasks

### PR DESCRIPTION
**Pull request by robfreedy**
This PR includes implementations for 4 pre-built Prefect DBT tasks. These pre-built tasks/flows will run the respective DBT command as a Prefect task and optionally will create a Prefect Markdown Artifact that provides a summary of the DBT runs, including any failed, successful, and skipped runs. 

Closes the following issues: 
https://github.com/PrefectHQ/prefect-dbt/issues/179
https://github.com/PrefectHQ/prefect-dbt/issues/180
https://github.com/PrefectHQ/prefect-dbt/issues/181
https://github.com/PrefectHQ/prefect-dbt/issues/182

### Example

```python
from prefect import flow
from prefect_dbt.cli.tasks import run_dbt_model, run_dbt_seed, run_dbt_snapshot, run_dbt_test


@flow
def dbt_flow():
    run_dbt_model(
        project_dir="/Users/test/prefect_dbt_project"
    )
    run_dbt_seed(
        project_dir="/Users/test/prefect_dbt_project"
    )
    run_dbt_snapshot(
        project_dir="/Users/test/prefect_dbt_project"
    )
    run_dbt_test(
        project_dir="/Users/test/prefect_dbt_project"
    )

if __name__ == "__main__":
    dbt_flow()

```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [x] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [x] This pull request includes helpful docstrings.
- [x] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: src/integrations/prefect-dbt/prefect_dbt/cli/flows.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from pathlib import Path  # noqa: I001
```
Comment (zzstoatzz) [inline src/integrations/prefect-dbt/prefect_dbt/cli/flows.py:1]: we should not need this `noqa` - we should follow `ruff`'s sorting

**Review by zzstoatzz — CHANGES_REQUESTED** [review]
at a glance, I'm thinking the names of these are pretty generic / unclear. i.e. are we mirroring the CLI here somehow?

can we include docstrings for all flows/tasks that have examples with valid imports? otherwise reading the API ref will be hard to follow

File: src/integrations/prefect-dbt/prefect_dbt/cli/tasks.py [hunk 2]
```diff
@@ -69,7 +73,326 @@ def dbt_test_flow():
...
+        dbt_client=dbt_client,
+        create_artifact=create_artifact,
+        artifact_key=artifact_key,
+        logger=logger,
+    )
+
+    return results
+
+
+def dbt_run(
```
Comment (desertaxle) [inline src/integrations/prefect-dbt/prefect_dbt/cli/tasks.py:379]: This `dbt_run` command appears to duplicate the existing `trigger_dbt_cli_command` function in this library. I'd prefer it if we reuse existing functionality in this library when building out this new interfaces. If the existing functionality doesn't support something that's needed for the new tasks and flows, let's chat and figure out a way to update the existing functionality to make it work for you. I want to prevent multiple implementations that will make it harder for users to navigate this library.

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: src/integrations/prefect-dbt/prefect_dbt/cli/flows.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        @flow
+        def dbt_test_flow():
+            dbt_flow(
+                project_dir="/Users/test/my_dbt_project_dir"
+            )
+    ```
+
+    Raises:
+        ValueError: If required dbt_cli_profile is not provided
```
Comment (discdiver) [inline src/integrations/prefect-dbt/prefect_dbt/cli/flows.py:60]: Line break formatting looks a little off in GitHub. Does it render ok?

**Review by discdiver — COMMENTED** [review]
Thank you! Added a few formatting nits/questions.

File: src/integrations/prefect-dbt/tests/cli/test_commands.py [hunk 4]
```diff
@@ -128,23 +185,25 @@ def test_trigger_dbt_cli_command_find_home(dbt_cli_profile_bare):
     @flow
     def test_flow():
         return trigger_dbt_cli_command(
-            "dbt ls", dbt_cli_profile=dbt_cli_profile, overwrite_profiles=False
+            "ls", dbt_cli_profile=dbt_cli_profile, overwrite_profiles=False
```
Comment (desertaxle) [inline src/integrations/prefect-dbt/tests/cli/test_commands.py:188]: For backwards compatibility, can we support passing in `dbt` as part of the command? We should be able to strip a leading `dbt` from the `command` before passing it to the `dbt_runner`.

File: src/integrations/prefect-dbt/prefect_dbt/cli/commands.py [hunk 5]
```diff
@@ -148,16 +145,47 @@ def trigger_dbt_cli_command_flow():
...
-    logger.info(f"Running dbt command: {command}")
-    result = await shell_run_command.fn(command=command, **shell_run_command_kwargs)
+    # fix up empty shell_run_command_kwargs
+    dbt_runner_client = dbtRunner()
+    logger.info(f"Running dbt command: {cli_args}")
+    result: dbtRunnerResult = dbt_runner_client.invoke(cli_args)
+
+    if result.exception is not None:
+        logger.error(f"dbt build task failed with exception: {result.exception}")
+        raise result.exception
```
Comment (desertaxle) [inline src/integrations/prefect-dbt/prefect_dbt/cli/commands.py:166]: Can you add a test covering this exception case?

**Review by desertaxle — CHANGES_REQUESTED** [review]
This is looking great! Just a couple final comments from me.

File: src/integrations/prefect-dbt/prefect_dbt/cli/commands.py [hunk 6]
```diff
@@ -148,16 +148,48 @@ def trigger_dbt_cli_command_flow():
...
 
-    logger.info(f"Running dbt command: {command}")
-    result = await shell_run_command.fn(command=command, **shell_run_command_kwargs)
+    # fix up empty shell_run_command_kwargs
+    dbt_runner_client = dbtRunner()
+    logger.info(f"Running dbt command: {cli_args}")
+    result: dbtRunnerResult = dbt_runner_client.invoke(cli_args)
+
+    if result.exception is not None:
+        logger.info(result)
```
Comment (desertaxle) [inline src/integrations/prefect-dbt/prefect_dbt/cli/commands.py:168]: This looks like a debug log that was left in since there's an error log directly below it?
```suggestion
```

**Review by desertaxle — APPROVED** [review]

**Review by zzstoatzz — APPROVED** [review]
